pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -176,52 +191,53 @@ func (f *FederatedTypeConfig) GetObjectMeta() metav1.ObjectMeta {
 	return f.ObjectMeta
 }
 
-func (f *FederatedTypeConfig) GetTarget() metav1.APIResource {
-	return apiResourceToMeta(f.Spec.Target, f.Spec.Namespaced)
+func (f *FederatedTypeConfig) GetTargetType() metav1.APIResource {
+	return apiResourceToMeta(f.Spec.TargetType, f.GetNamespaced())
 }
 
 func (f *FederatedTypeConfig) GetNamespaced() bool {
```

Comment:
This function should go away (i.e. be removed from the interface) in favor of checking the namespaced property of the appropriate apiresource (federated or target depending on the context).

Reply:
Please capture this as a TODO on the issue so it doesn't get misplaced.